tests/logger/test_log_entry.py

```diff
@@ -1101,601 +1103,946 @@ def test_deserialization_empty_fails(self, empty_set_signal_log_entry_as_dict):
-                "value_after": value_after,
+                "train_id": train_id,
+                "block_section_id": block_section_id,
+                "block_section_length": block_section_length,
             }
 
         @pytest.fixture
-        def empty_inject_fault_log_entry_as_dict(self):
-            """InjectFaultLogEntry as dict with no fields set."""
+        def empty_train_enter_block_section_log_entry_as_dict(self):
+            """TrainEnterBlockSectionLogEntry as dict with no fields set."""
             return {}
 
         @recreate_db_setup
         def setup_method(self):
             pass
 
-        def test_create(self, inject_fault_log_entry_as_dict):
-            """Test that Inject Fault Log Entry can be created."""
-            inject_fault_log_entry = InjectFaultLogEntry.create(
-                **inject_fault_log_entry_as_dict
+        def test_create(self, train_enter_block_section_log_entry_as_dict):
+            """Test that TrainEnterBlockSectionLogEntry can be created."""
+            train_enter_block_section_log_entry = TrainEnterBlockSectionLogEntry.create(
+                **train_enter_block_section_log_entry_as_dict
             )
             assert (
-                InjectFaultLogEntry.select()
-                .where(InjectFaultLogEntry.id == inject_fault_log_entry.id)
+                TrainEnterBlockSectionLogEntry.select()
+                .where(
+                    TrainEnterBlockSectionLogEntry.id
+                    == train_enter_block_section_log_entry.id
+                )
                 .first()
-                == inject_fault_log_entry
+                == train_enter_block_section_log_entry
             )
 
-        def test_create_empty_fails(self, empty_inject_fault_log_entry_as_dict):
-            """Test that Inject Fault Log Entry cannot be created with no fields set."""
+        def test_create_empty_fails(
+            self, empty_train_enter_block_section_log_entry_as_dict
+        ):
+            """Tests that TrainEnterBlockSectionLogEntry cannot be created with no fields set."""
             with pytest.raises(IntegrityError):
-                InjectFaultLogEntry.create(**empty_inject_fault_log_entry_as_dict)
+                TrainEnterBlockSectionLogEntry.create(
+                    **empty_train_enter_block_section_log_entry_as_dict
+                )
 
-        def test_serialization(self, inject_fault_log_entry_as_dict):
-            """Test that Inject Fault Log Entry can be serialized."""
-            inject_fault_log_entry = InjectFaultLogEntry.create(
-                **inject_fault_log_entry_as_dict
+        def test_serialization(self, train_enter_block_section_log_entry_as_dict):
+            """Test that TrainEnterBlockSectionLogEntry can be serialized."""
+            train_enter_block_section_log_entry = TrainEnterBlockSectionLogEntry.create(
+                **train_enter_block_section_log_entry_as_dict
             )
```

Comment:
@Lietze and @CR1337 told me yesterday for a similar test of me, that for testing the serialization a `to_dict` is missing. What is correct now?

Reply:
nervermind, you test that below :)